servicelog/appender/logstash.go

```diff
@@ -107,14 +120,46 @@ func NewLogstash(writer io.Writer, options ...func(*logstash) error) (Appender,
 	return l, nil
 }
 
+// NewConsulLogstashWriter creates a new writer that will write data to instances
+// provided by local Consul agent. It will use round robin algorithm to spread
+// logs evenly to every Logstash instance.
+func NewConsulLogstashWriter(protocol, serviceName string, refreshInterval time.Duration, dialer net.Dialer) (io.Writer, error) {
```

Comment:
How about removing protocol and 

```go
if dialer == nil {
	sender = &xnet.UDPSender{}
} else {
	tcpSender := &xnet.TCPSender{
		Dialer: dialer,
	}
	sender = tcpSender
}
```

Reply:
I think it will be more transparent when the protocol will be passed explicitly - especially that `net.Dialer` can be used for UDP and TCP dialling.